src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -423,14 +406,39 @@ private bool TryDequeueRenderQueueEntry(out RenderQueueEntry entry)
             }
         }
 
-        private void InvokeRenderCompletedCalls(ArrayRange<RenderTreeDiff> updatedComponents)
+        private async void InvokeRenderCompletedCalls(ArrayRange<RenderTreeDiff> updatedComponents)
         {
+            List<Task> batch = null;
             var array = updatedComponents.Array;
             for (var i = 0; i < updatedComponents.Count; i++)
             {
-                // The component might be rendered and disposed in the same batch (if its parent
-                // was rendered later in the batch, and removed the child from the tree).
-                GetOptionalComponentState(array[i].ComponentId)?.NotifyRenderCompleted();
+                var componentState = GetOptionalComponentState(array[i].ComponentId);
+                if (componentState != null)
+                {
+                    // The component might be rendered and disposed in the same batch (if its parent
+                    // was rendered later in the batch, and removed the child from the tree).
+                    batch = batch ?? new List<Task>();
+                    batch.Add(NotifyRenderCompletedAsync(componentState));
+                }
+            }
+
+            if (batch != null)
+            {
+                await Task.WhenAll(batch);
```

Comment:
Mimics the previous behavior where `NotifyRenderCompleted` was an `async void` method